ansible/roles/jenkins-node/tasks/main.yaml

```diff
@@ -0,0 +1,60 @@
+- name: Create jenkins group.
+  group:
+    name: jenkins
+
+- name: Create jenkins user.
+  user:
+    createhome: yes
+    group: jenkins
+    home: /home/jenkins
+    name: jenkins
+
+- name: Make .ssh dir for jenkins
+  file:
+    path: /home/jenkins/.ssh
+    state: directory
+    mode: 0700
+    owner: jenkins
+    group: jenkins
+
+- name: Install ssh key for jenkins
+  authorized_key:
+    key: "{{ lookup('file', '{{ jenkins_public_key }}' ) }}"
+    state: present
+    user: jenkins
+
+- name: Allow jenkins to run sudo commands without password.
+  lineinfile:
+    path: /etc/sudoers
+    state: present
+    regexp: '^%jenkins\s'
+    line: '%jenkins ALL=(ALL) NOPASSWD: ALL'
+
+- name: Install custom repo file
+  copy:
+    src: "{{ custom_repo_file }}"
+    dest: /etc/yum.repos.d/custom.repo
+
+- name: Install epel
```

Comment:
epel is only needed for Red Hat Enterprise Linux, also you can figure out the url based on the OS major version. Please make this run only on RHEL.